vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -262,37 +269,45 @@ facebook::jsi::Object ChakraRuntime::createObject() {
 }
 
 facebook::jsi::Object ChakraRuntime::createObject(std::shared_ptr<facebook::jsi::HostObject> hostObject) {
-  facebook::jsi::Object proxyTarget =
-      ObjectWithExternalData<HostObjectProxy>::create(*this, std::make_unique<HostObjectProxy>(*this, hostObject));
-  return createProxy(std::move(proxyTarget), createHostObjectProxyHandler());
+  constexpr const char *const jsProxyConstructorWrapperSource =
+      "(function()\n"
+      "{\n"
+      "  return function(target, handler)\n"
+      "  {\n"
+      "    return new Proxy(target, handler)\n"
+      "  }\n"
+      "})()";
```

Comment:
If we'll end up having a bunch of these snippets, would it be worth creating a "bootstrap.bundle" that we load in each chakraruntime post-init? At least we'd only interpret this code once (and maybe give JIT a chance to optimize it into native).

Reply:
I think that is a good idea. This snippet and the one in getPropertyNames are the only ones at the moment, but I suspect that we might need another JS snippet if we want to implement getHostFunction.

---
In reply to: [338808639](https://github.com/microsoft/react-native-windows/pull/3514#discussion_r338808639) [](ancestors = 338808639)